### PR TITLE
added check to detect empty connection info struct

### PR DESCRIPTION
**Pull request by rtpa25**
## Summary of change

Added an if Check to verify and throw an error when user passes connectionInfo as an empty struct 

## Related issues

-   No isue was created for this PR

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: supertokens/supertokens.go [hunk 1]
```diff
@@ -54,16 +54,20 @@ func supertokensInit(config TypeInput) error {
...
+				host, err := NewNormalisedURLDomain(h)
+				if err != nil {
+					return err
+				}
+				hosts = append(hosts, host)
 			}
-			hosts = append(hosts, host)
+			initQuerier(hosts, config.Supertokens.APIKey)
+		} else {
+			return errors.New("please provide 'ConnectionURI' value")
```
Comment (rishabhpoddar) [inline supertokens/supertokens.go:69]: Change message to `"please provide 'ConnectionURI' value. If you do not want to provide a connection URI, then set config.Supertokens to nil"`

File: supertokens/supertokens.go [hunk 2]
```diff
@@ -54,16 +54,20 @@ func supertokensInit(config TypeInput) error {
 	}
 
 	if config.Supertokens != nil {
-		hostList := strings.Split(config.Supertokens.ConnectionURI, ";")
-		var hosts []NormalisedURLDomain
-		for _, h := range hostList {
-			host, err := NewNormalisedURLDomain(h)
-			if err != nil {
-				return err
+		if len(config.Supertokens.ConnectionURI) != 0 {
```
Comment (rishabhpoddar) [inline supertokens/supertokens.go:57]: You need to change changelog

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens/supertokens.go (hunk 1 above)
Comment (rtpa25) [reply supertokens/supertokens.go]: Updated the error message

File: supertokens/supertokens.go (hunk 2 above)
Comment (rtpa25) [reply supertokens/supertokens.go:57]: added relevant change message to the changelog file

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 -   Adds passwordless recipe
 -   Adds compatibility with FDI 1.11 and CDI 2.11
+-   Adds check to see if user has provided empty connectionInfo
```
Comment (rishabhpoddar) [inline CHANGELOG.md:14]: Move to unreleased

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**